### PR TITLE
Add Windows build + functional test (using nmake)

### LINKED ISSUE #227: Add support for and test Windows platforms

### PR DESCRIPTION
**Pull request by potsrevennil**
- Resolves #227
- For simplicity, I only provide a `quickcheck` target in the windows makefile for building and running the functional tests binaries
- Windows nmake command:
  - `nmake /f ./Makefile.Microsoft_nmake quickcheck`
- Built binaries path:
  - `test/build/mlkem512/bin/test_mlkem512.exe`
  - `test/build/mlkem768/bin/test_mlkem768.exe`
  - `test/build/mlkem1024/bin/test_mlkem1024.exe`
- Only support for compiling non-opt implementation, as I don't know how to make the assembly compatible on Windows

[//]: # (SPDX-License-Identifier: CC-BY-4.0)

### CODE REVIEW

File: fips202/keccakf1600.h [hunk 1]
```diff
@@ -16,7 +16,7 @@
 //
 // The struct is only exposed here to allow its construction on the stack.
 //
-typedef uint64_t keccakx4_state[KECCAK_WAY * KECCAK_LANES] ALIGN;
+typedef ALIGN uint64_t keccakx4_state[KECCAK_WAY * KECCAK_LANES];
```
Comment (hanno-becker) [inline fips202/keccakf1600.h:19]: The movement of `ALIGN` should be explained in the commit message, and be done in a separate commit.

File: Makefile.Microsoft_nmake [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CFLAGS = /nologo /O2 /Imlkem /Ifips202 /Ifips202/native /Imlkem/sys /Imlkem/native
```
Comment (hanno-becker) [inline Makefile.Microsoft_nmake:3]: Does this build opt or non-opt, and (how) can one switch between the two?

**Review by hanno-becker — CHANGES_REQUESTED** [review]
Thank you @potsrevennil, it's great to see progress here!

A few points in the comments. Also, can you please expand the README to explain how to build on Windows?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -49,6 +49,18 @@ jobs:
       - name: check namespacing
         run: |
           ./scripts/ci/check-namespace
+  quickcheck-windows:
+    name: Quickcheck windows-latest
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: ilammy/msvc-dev-cmd@v1
```
Comment (hanno-becker) [inline .github/workflows/ci.yml:57]: How configurable is this? Can we test different MSVC versions?

File: Makefile.Microsoft_nmake [hunk 4]
```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (hanno-becker) [inline Makefile.Microsoft_nmake:1]: Is this makefile specific to MSVC? Which versions of MSVC would it work with?

File: Makefile.Microsoft_nmake (hunk 2 above)
Comment (potsrevennil) [reply Makefile.Microsoft_nmake:3]: I currently only compile non-opt code, as I don't know how to compile the assembly files on windows.

For example I get a bunch of errors like
```
.\mlkem\native\x86_64\basemul.S(1) : error A2008:syntax error : /
.\mlkem\native\x86_64\basemul.S(3) : error A2008:syntax error : /
.\mlkem\native\x86_64\basemul.S(4) : error A2008:syntax error : /
.\mlkem\native\x86_64\basemul.S(6) : error A2044:invalid character in file
.\mlkem\native\x86_64\basemul.S(7) : error A2044:invalid character in file
.\mlkem\native\x86_64\basemul.S(8) : error A2044:invalid character in file
.\mlkem\native\x86_64\basemul.S(9) : error A2044:invalid character in file
...
```
when trying to compile assembly files to object file by
```
ml /c .\mlkem\native\x86_64\basemul.S
```

File: .github/workflows/ci.yml (hunk 3 above)
Comment (potsrevennil) [reply .github/workflows/ci.yml:57]: We seemed to be able to test a bunch of different set of MSVC configuration

https://github.com/ilammy/msvc-dev-cmd?tab=readme-ov-file#inputs

> arch – target architecture
> - native compilation:
>   - x64 (default) or its synonyms: amd64, win64, x86_64, x86-64
>   - x86 or its synonyms: win32
> cross-compilation: x86_amd64, x86_arm, x86_arm64, amd64_x86, amd64_arm, amd64_arm64
> sdk – Windows SDK to use
> - do not specify to use the default SDK
> - or specify full Windows 10 SDK number (e.g, 10.0.10240.0)
> - or write 8.1 to use Windows 8.1 SDK
> toolset – select VC++ compiler toolset version
> - do not specify to use the default toolset
> - 14.0 for VC++ 2015 Compiler Toolset
> - 14.XX for the latest 14.XX toolset installed (e.g, 14.11)
> - 14.XX.YYYYY for a specific full version number (e.g, 14.11.25503)
> uwp – set true to build for Universal Windows Platform (i.e., for Windows Store)
> spectre – set true to use Visual Studio libraries with [Spectre](https://meltdownattack.com/) mitigations
> vsversion – the Visual Studio version to use. This can be the version number (e.g. 16.0 for 2019) or the year (e.g. "2019"); omit this input to select the latest version of Visual Studio. On [GitHub-hosted runners](https://docs.github.com/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners), this input is not required because there is only one Visual Studio in the environment.

File: Makefile.Microsoft_nmake (hunk 4 above)
Comment (potsrevennil) [reply Makefile.Microsoft_nmake:1]: Yes, this is specific to MSVC, and I currently only test with this version, 
```
Microsoft (R) C/C++ Optimizing Compiler Version 19.42.34432 for x86
```

File: Makefile.Microsoft_nmake (hunk 2 above)
Comment (mkannwischer) [reply Makefile.Microsoft_nmake:3]: Let's restrict to the non-opt implementation for this PR - please write it somewhere as a comment in the Makefile. 
I'll try to find someone who can help us with getting the assembly implementations to work.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (hanno-becker) [reply .github/workflows/ci.yml:57]: Great! Could you make it so the CI tests with an older version as well?

File: Makefile.Microsoft_nmake (hunk 4 above)
Comment (hanno-becker) [reply Makefile.Microsoft_nmake:1]: Can you leave a comment to this end in the commit message and in the Makefile? (Hopefully adding an older version as well, see other comment)

File: Makefile.Microsoft_nmake (hunk 2 above)
Comment (hanno-becker) [reply Makefile.Microsoft_nmake:3]: Works for me, but it should be commented on in the Makefile, README, and commit message.

File: fips202/keccakf1600.h (hunk 1 above)
Comment (hanno-becker) [reply fips202/keccakf1600.h:19]: Ah ok, the `;` separates things cleanly still. So it may well be that gcc/clang are agnostic to the placement.

**Review by hanno-becker — APPROVED** [review]
I got a bit pedantic. @potsrevennil This is a great start, thank you very much.
